File: ru.kmorozov.gbd.core.config/src/main/java/ru/kmorozov/gbd/core/config/options/PdfOptions.java
package ru.kmorozov.gbd.core.config.options;

public enum PdfOptions {
    CREATE("PDF_CREATE"),
    SKIP("PDF_SKIP");

    private final String pdfMode;
    public static final PdfOptions DEFAULT_MODE = PdfOptions.CREATE;

    PdfOptions(String pdfMode) {
        this.pdfMode = pdfMode;
    }

    public static PdfOptions getOption(String pdfMode) {
        for(PdfOptions option : PdfOptions.values())
            if (option.pdfMode.equalsIgnoreCase(pdfMode))
                return option;

        return PdfOptions.DEFAULT_MODE;
    }
}

File: ru.kmorozov.gbd.core.config/src/main/java/ru/kmorozov/gbd/core/config/options/PdfOptions.kt
package ru.kmorozov.gbd.core.config.options

enum class PdfOptions(private val pdfMode: String) {
    CREATE("PDF_CREATE"),
    CREATE_ONLY("PDF_CREATE_ONLY"),
    SKIP("PDF_SKIP");

    companion object {
        val DEFAULT_MODE = CREATE

        fun getOption(pdfMode: String): PdfOptions {
            for (option in values())
                if (option.pdfMode.equals(pdfMode, ignoreCase = true))
                    return option

            return DEFAULT_MODE
        }
    }
}
